src/nvim/eval.c

```diff
@@ -8138,13 +8133,85 @@ void ex_execute(exarg_T *eap)
   eap->nextcmd = check_nextcmd(arg);
 }
 
-/// Skip over the name of an option: "&option", "&g:option" or "&l:option".
+/// Skip over the name of a TTY option or keycode option.
+///
+/// @param[in]  arg  Start of TTY or keycode option name.
+///
+/// @return NULL when option isn't a TTY or keycode option. Otherwise pointer to the char after the
+/// option name.
+const char *find_tty_option_end(const char *arg)
+{
+  if (strequal(arg, "term")) {
+    return arg + sizeof("term") - 1;
+  } else if (strequal(arg, "ttytype")) {
+    return arg + sizeof("ttytype") - 1;
+  }
+
+  const char *p = arg;
+  bool delimit = false;  // whether to delimit <
+
+  if (arg[0] == '<') {
+    // look out for <t_>;>
+    delimit = true;
+    p++;
+  }
+  if (p[0] == 't' && p[1] == '_' && p[2] && p[3]) {
+    p += 4;
+  } else if (delimit) {
+    // Search for delimiting >.
+    while (*p != NUL && *p != '>') {
+      p++;
+    }
+  }
+  // Return NULL when delimiting > is not found.
+  if (delimit) {
+    if (*p != '>') {
+      return NULL;
+    }
+    p++;
+  }
+
+  return arg == p ? NULL : p;
+}
+
+/// Skip over the name of an option.
+///
+/// @param[in]   arg       Start of option name.
+/// @param[out]  opt_idxp  Set to option index in options[] table.
+///
+/// @return NULL when no option name found. Otherwise pointer to the char after the option name.
+const char *find_option_end(const char *arg, OptIndex *opt_idxp)
+{
+  const char *p;
+
+  // Handle TTY and keycode options separately.
+  if ((p = find_tty_option_end(arg)) != NULL) {
+    *opt_idxp = kOptInvalid;
+    return p;
+  } else {
+    p = arg;
+  }
+
+  if (!ASCII_ISALPHA(*p)) {
+    *opt_idxp = kOptInvalid;
+    return NULL;
+  }
+  while (ASCII_ISALPHA(*p)) {
+    p++;
+  }
+
+  *opt_idxp = findoption_len(arg, (size_t)(p - arg));
+  return p;
+}
```

Comment:
These two functions should now be moved to option.c. Then `find_tty_option_end()` can be static.